docs/topics/dokka-migration.md

```diff
@@ -77,157 +95,355 @@ After updating the Dokka Gradle plugin to v2, follow the migration steps applica
 
-  Additionally, DGP v2 has a [utility function](https://github.com/Kotlin/dokka/blob/220922378e6c68eb148fda2ec80528a1b81478c9/dokka-runners/dokka-gradle-plugin/src/main/kotlin/engine/parameters/HasConfigurableVisibilityModifiers.kt#L14-L16) for adding documented visibilities:
- 
-  ```kotlin
-  fun documentedVisibilities(vararg visibilities: VisibilityModifier): Unit =
-      documentedVisibilities.set(visibilities.asList()) 
-  ```
+// ...
+documentedVisibilities.set(
+    setOf(Visibility.PUBLIC)
+) 
+```
 
-* **External documentation link:** The source link configuration has [changed](https://docs.gradle.org/current/userguide/upgrading_version_8.html#deprecated_invalid_url_decoding). 
-    The remote URL is now specified using the `URI` class instead of the `URL` class.
+New configuration:
 
-  Previous configuration:
-  
-    ```kotlin
-    remoteUrl.set(URL("https://github.com/your-repo"))
-    ```
-  
-  New configuration:
-  
-    ```kotlin
-    remoteUrl.set(URI("https://github.com/your-repo"))
-    
-    // OR
-    
-    remoteUrl("https://github.com/your-repo")
-    ```
+```kotlin
+import org.jetbrains.dokka.gradle.engine.parameters.VisibilityModifier
+
+// ...
+documentedVisibilities.set(
+    setOf(VisibilityModifier.Public)
+)
+
+// OR
+
+documentedVisibilities(VisibilityModifier.Public)
+```
+
+Additionally, DGP v2 has a [utility function](https://github.com/Kotlin/dokka/blob/220922378e6c68eb148fda2ec80528a1b81478c9/dokka-runners/dokka-gradle-plugin/src/main/kotlin/engine/parameters/HasConfigurableVisibilityModifiers.kt#L14-L16) for adding documented visibilities:
+
+```kotlin
+fun documentedVisibilities(vararg visibilities: VisibilityModifier): Unit =
+    documentedVisibilities.set(visibilities.asList()) 
+```
 
-  Additionally, DGP v2 has two [utility functions](https://github.com/Kotlin/dokka/blob/220922378e6c68eb148fda2ec80528a1b81478c9/dokka-runners/dokka-gradle-plugin/src/main/kotlin/engine/parameters/DokkaSourceLinkSpec.kt#L82-L96) 
-  for setting the URL:
+#### Source links
 
-  ```kotlin
-  fun remoteUrl(@Language("http-url-reference") value: String): Unit =
-      remoteUrl.set(URI(value))
+Allow users to navigate from the generated documentation to the corresponding source code in a remote repository. 
+To configure source links, use the `dokkaSourceSets.main{}` block.
+
+Previous configuration:
+
+```kotlin
+    tasks.withType<DokkaTask>().configureEach {
+    dokkaSourceSets {
+        named("main") {
+            sourceLink {
+                localDirectory.set(file("src/main/kotlin"))
+                remoteUrl.set(URL("https://github.com/your-repo"))
+                remoteLineSuffix.set("#L")
+            }
+        }
+    }
+}
+```
+
+New configuration:
+
+```kotlin
+    dokka {
+    moduleName.set("Project Name")
+    dokkaSourceSets.main {
+        includes.from("README.md")
+        sourceLink {
+            localDirectory.set(file("src/main/kotlin"))
+            remoteUrl("https://github.com/your-repo")
```

Comment:
For consistency (`set` in all properties), maybe we could use `remoteUrl.set(URI("https://github.com/your-repo"))`.